tests/unit/test_pyproject_config.py

```
@@ -0,0 +1,352 @@
+        # Should be at least 3.11
+        major = int(version_parts[0])
+        minor = int(version_parts[1])
+
+        assert major >= 3, "Python major version must be at least 3"
+        assert major == 3 and minor >= 11, (
+            f"Python version must be at least 3.11, got {version_str}"
+        )
+
+    def test_toml_syntax_validation(self, pyproject_path: pathlib.Path) -> None:
+        """Test that pyproject.toml has valid TOML syntax."""
+        import tomllib
```

Ruff (0.12.2)

`import` should be at the top-level of a file

(PLC0415)